internal/controller/api/v1alpha1/installenv_types.go

```diff
@@ -0,0 +1,111 @@
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	corev1 "k8s.io/api/core/v1"
+)
+
+type InstallEnvSpec struct {
+	// Proxy defines the proxy settings for agents and clusters that use the
+	// InstallEnv. If unset, the agents and clusters will not be configured to use a
+	// proxy.
+	// +optional
+	Proxy *Proxy `json:"proxy,omitempty"`
+
+	// AdditionalNTPSources is a list of NTP sources (hostname or IP) to be added to 
+	// all cluster hosts. They are added to any NTP sources that were configured through
+	// other means.
+	// +optional
+	AdditionalNTPSources []string `json:"additionalNTPSources,omitempty"`
+
+	// SSHAuthorizedKeys is a list of SSH public keys that will be added to all agents 
+	// for use in debugging.
+	// +optional
+	SSHAuthorizedKeys []string `json:"sshAuthorizedKeys,omitempty"`
+
+	// PullSecretRef is the reference to the secret to use when pulling images.
+	PullSecretRef *corev1.LocalObjectReference `json:"pullSecretRef"`
+
+	// Agents that boot as a result of this InstallEnv should be labeled such that
+	// they match this selector. That is how a user would identify which agents are
+	// associated with a particular InstallEnv.
+	AgentLabelSelector metav1.LabelSelector `json:"agentLabelSelector"`
+
+	// Any labels to apply to agents at creation time that are associated with this 
+	// InstallEnv.
+	// +optional
+	AgentLabels map[string]string `json:"agentLabels,omitempty"`
+
+	// NmstateConfigs that match this label are considered part of this InstallEnv.
+	// +optional
+	NmstateConfigLabelSelector metav1.LabelSelector `json:"agentLabelSelector"`
+
+	// ClusterRef is the reference to the single ClusterDeployment that will 
+	// be installed from this InstallEnv.
+	// Future versions will allow for multiple ClusterDeployments and this reference 
+	// will be removed.
+	ClusterRef   *ClusterReference `json:"clusterRef"`
+}
+
+
+// Proxy defines the proxy settings for agents and clusters that use the InstallEnv.
+// At least one of HTTPProxy or HTTPSProxy is required.
+type Proxy struct {
+	// HTTPProxy is the URL of the proxy for HTTP requests.
+	// +optional
+	HTTPProxy string `json:"httpProxy,omitempty"`
+
+	// HTTPSProxy is the URL of the proxy for HTTPS requests.
+	// +optional
+	HTTPSProxy string `json:"httpsProxy,omitempty"`
+
+	// NoProxy is a comma-separated list of domains and CIDRs for which the proxy should
+	// not be used.
+	// +optional
+	NoProxy string `json:"noProxy,omitempty"`
+}
+
+type InstallEnvStatus struct {
+	DownloadURL string `json:"downloadURL,omitempty"`
```

Comment:
We talked about also having a field for the generation the ISO is based on.

Reply:
Also, maybe we should call this `ISODownloadURL` to make it clear what you can download.

Reply:
I am going to unhold so that we can proceed with the implementation - @mhrivnak let's discuss and I can add to another PR if necessary